stubs/nette-di.stub

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Nette\DI;
+
+/**
+ * The dependency injection container default implementation.
+ */
+class Container
+{
+	/**
+	 * @phpstan-template T of object
+	 * @phpstan-param class-string<T> $type
+	 * @phpstan-return T|null
```

Comment:
I forgot there's already an extension for this method: https://github.com/phpstan/phpstan-src/blob/master/build/PHPStan/Build/ServiceLocatorDynamicReturnTypeExtension.php

So we actually don't need a stub, but to fix this extension. I did it on master: https://github.com/phpstan/phpstan-src/commit/12973cdb1c3c0f84e09dd039edfbf66a58991878

So you can rebase your branch, remove the stub and any unrelated changes, and it should be fine, hopefully.